src/client/helpers/metadata.js

```diff
@@ -1,75 +1,85 @@
 import omit from 'lodash/omit';
 import SteemConnect from '../steemConnectAPI';
 
-const getMetadata = () => SteemConnect.me().then(resp => resp.user_metadata);
+// sc2 metadata support discontinuation preparation
+// const getMetadata = () => SteemConnect.me().then(resp => resp.user_metadata);
+export const getMetadata = () =>
+  SteemConnect.me().then(resp => {
+    const localMetadata = JSON.parse(localStorage.getItem('user_metadata'));
+    if (localMetadata) {
+      return Promise.resolve(localMetadata);
+    }
+
+    try {
+      localStorage.setItem('user_metadata', JSON.stringify(resp.user_metadata));
+      return resp.user_metadata;
+    } catch (err) {
+      return Promise.reject(new Error(err));
+    }
+  });
+
+export const updateUserMetadata = metadata => {
+  localStorage.setItem('user_metadata', JSON.stringify(metadata));
+  return Promise.resolve(metadata);
```

Comment:
You dont need a promise anymore for this method

Reply:
I know but one problem is all the other codes later on expect promise, so i had to choose this dirty trick rather than rewriting so many parts.

Thanks a lot for your review!

ps. Any idea on the known issues that I wrote?